test/view_component/view_component_test.rb

```diff
@@ -61,6 +61,14 @@ def test_renders_haml_template
     assert_text("bar")
   end
 
+  def test_renders_haml_template_that_uses_a_block
+    render_inline(HamlComponent.new(message: "bar", url: "http://example.com")) { "foo" }
+
+    assert_selector("a[href='http://example.com']", "hello!")
```

Comment:
```suggestion
    assert_selector("a[href='http://example.com']", text: "hello!")
```

Once my previous suggestion is applied, this line will cause the following error:

```
Error:
ViewComponentTest#test_renders_haml_template_that_uses_a_block:
ArgumentError: Unused parameters passed to Capybara::Queries::SelectorQuery : ["hello!"]
```

This suggestion fixes the issue.